pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -1416,18 +1416,21 @@ public CompletableFuture<TopicExistsInfo> checkTopicExists(TopicName topic) {
         return pulsar.getBrokerService().fetchPartitionedTopicMetadataAsync(
                         topic.isPartitioned() ? TopicName.get(topic.getPartitionedTopicName()) : topic)
                 .thenCompose(metadata -> {
+                    // When the topic has metadata:
+                    // - The topic name is non-partitioned, which means that the topic exists.
+                    // - The topic name is partitioned, please check the specific partition.
                     if (metadata.partitions > 0) {
                         if (!topic.isPartitioned()) {
                             return CompletableFuture.completedFuture(
                                     TopicExistsInfo.newPartitionedTopicExists(metadata.partitions));
-                        } else {
-                            if (topic.getPartitionIndex() < metadata.partitions) {
-                                return CompletableFuture.completedFuture(
-                                        TopicExistsInfo.newNonPartitionedTopicExists());
-                            }
+                        }
+                        // The non-persistent partitioned topic doesn't have the partition, quickly return.
```

Comment:
What does this comment mean? more specifically, what does "doesn't have the partition" mean?

Reply:
@lhotari The non-persistent partitioned topic only has the metadata.

Reply:
Making this comment more understandable would be helpful for future. Using Co-pilot in IntelliJ is one way to address this. Just ask it to improve the code comment and provide the above review comments as context.

Reply:
Done.